tests/test_reqrep_tcp.cpp

```diff
@@ -28,129 +28,77 @@
 */
 
 #include "testutil.hpp"
+#include "testutil_unity.hpp"
 
-void test_single_connect (const char *address)
+void setUp ()
+{
+    setup_test_context ();
+}
+
+void tearDown ()
+{
+    teardown_test_context ();
+}
+
+void test_single_connect (int ipv6)
 {
     size_t len = MAX_SOCKET_STRING;
     char my_endpoint[MAX_SOCKET_STRING];
-    void *ctx = zmq_ctx_new ();
-    assert (ctx);
-
-    int ipv6;
-    if (streq (address, "tcp://127.0.0.1:*"))
-        ipv6 = 0;
-    else if (streq (address, "tcp://[::1]:*"))
-        ipv6 = 1;
-    else
-        assert (false);
-
-    if (ipv6 && !is_ipv6_available ()) {
-        zmq_ctx_term (ctx);
-        return;
-    }
-
-    void *sb = zmq_socket (ctx, ZMQ_REP);
-    assert (sb);
-    int rc = zmq_setsockopt (sb, ZMQ_IPV6, &ipv6, sizeof (int));
-    assert (rc == 0);
-    rc = zmq_bind (sb, address);
-    assert (rc == 0);
-    rc = zmq_getsockopt (sb, ZMQ_LAST_ENDPOINT, my_endpoint, &len);
-    assert (rc == 0);
-
-    void *sc = zmq_socket (ctx, ZMQ_REQ);
-    assert (sc);
-    rc = zmq_setsockopt (sc, ZMQ_IPV6, &ipv6, sizeof (int));
-    assert (rc == 0);
-    rc = zmq_connect (sc, my_endpoint);
-    assert (rc == 0);
+
+    void *sb = test_context_socket (ZMQ_REP);
+    bind_loopback (sb, ipv6, my_endpoint, len);
+
+    void *sc = test_context_socket (ZMQ_REQ);
+    TEST_ASSERT_SUCCESS_ERRNO (
+      zmq_setsockopt (sc, ZMQ_IPV6, &ipv6, sizeof (int)));
+    TEST_ASSERT_SUCCESS_ERRNO (zmq_connect (sc, my_endpoint));
 
     bounce (sb, sc);
 
-    rc = zmq_disconnect (sc, my_endpoint);
-    assert (rc == 0);
+    // TODO is explicit disconnect/unbind essential for the test? if not,
```

Comment:
yes - it's checking that endpoints are saved correctly, especially for v6 there were a few regressions in the past were the strings didn't match

Reply:
Ok, then I will change this to a comment that explains this.